site/src/lib/model/user.model.ts

```diff
@@ -220,19 +263,30 @@ export class User {
     const numberOfRecentLoginCodes = await db
       .collection<VerificationCodeDocument>(VerificationCode.COLLECTION_NAME)
       .count({
-        _id: { $in: this.loginCodes.map(({ oid }) => oid) },
-        createdAt: {
-          $gt: new Date(
-            new Date().getTime() - User.LOGIN_CODE_RATE_LIMIT_PERIOD_MS,
-          ),
-        },
+        $and: [
+          {
+            user: this.toRef(),
+          },
+          {
+            variant: "login",
+          },
+          {
+            createdAt: {
+              $gt: new Date(
+                new Date().getTime() - User.LOGIN_CODE_RATE_LIMIT_PERIOD_MS,
+              ),
+            },
+          },
+        ],
```

Comment:
You can express conditions like this (require multiple key-value pairs on a document) as a single object, but in practice it's probably optimized to make no difference except to the code. 

```TypeScript
          {
            user: this.toRef(),
            variant: "login",
            createdAt: {
              $gt: new Date(
                new Date().getTime() - User.LOGIN_CODE_RATE_LIMIT_PERIOD_MS,
              ),
            },
          },
```